### PR TITLE
Be more precise for `<<` of Intervals

### LINKED ISSUE #1250: Be more precise for overflow warnings for shifts
For the following program we produce a warning that we might have a signed integer over-and underflow. This seems unnecessarily imprecise.

~~~C
// PARAM: --enable ana.int.interval
int main()
{
    int r;
    unsigned char c;
    r = c << 1U;
    return 0;
}
~~~

### PR DESCRIPTION
**Pull request by michael-schwarz**
Closes #1250

### CODE REVIEW

**Comment by michael-schwarz** [comment]
This also means we succeed on the relevant SV-COMP task.

**Comment by michael-schwarz** [comment]
Looks .like there's a few more SV-COMP No-Overflow tasks where this helps :)

**Comment by michael-schwarz** [comment]
Fails on some LDV tasks e.g.

>./goblint --conf conf/svcomp24.json --sets ana.specification /home/goblint2/svcomp/sv-benchmarks/c/properties/unreach-call.prp --sets exp.architecture 64bit /home/goblint2/svcomp/sv-benchmarks/c/ldv-linux-3.0/usb_urb-drivers-video-arkfb.ko.cil.out.i

with

> Fatal error: exception Invalid_argument("Z.shift_left: count argument must be positive")

File: src/cdomains/intDomain.ml [hunk 1]
```diff
@@ -610,6 +610,11 @@ module IntervalArith(Ints_t : IntOps.IntOps) = struct
     let x2y2 = (Ints_t.mul x2 y2) in
     (min4 x1y1 x1y2 x2y1 x2y2, max4 x1y1 x1y2 x2y1 x2y2)
 
+  let shiftleft (x1,x2) (y1,y2) =
```
Comment (sim642) [inline src/cdomains/intDomain.ml:613]: Everywhere else this is named `shift_left`.

**Comment by michael-schwarz** [comment]
Since this has an overall positive effect, I suggest to merge.

File: tests/regression/39-signed-overflows/06-shiftleft.c [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    if(top) { zero_or_one = 1; }
+
+    r = 8 << zero_or_one;
+
+    __goblint_check(r >= 8);
+    __goblint_check(r <= 16);
+
+    int regval;
+    unsigned long bla = (unsigned long )((1 << ((int )regval >> 6)) << 20);  //WARN
```
Comment (jerhard) [inline tests/regression/39-signed-overflows/06-shiftleft.c:23]: Doesn't the subexpression `(1 << ((int )regval >> 6)` already overflow potentially? If so, one may add a line for that.

**Review by sim642 — APPROVED** [review]

File: tests/regression/39-signed-overflows/06-shiftleft.c (hunk 2 above)
Comment (sim642) [reply tests/regression/39-signed-overflows/06-shiftleft.c:23]: Due to #1258, this might only be warning due to the cast right now?

Comment (jerhard) [reply tests/regression/39-signed-overflows/06-shiftleft.c]: The subexpression actually already potentially overflows and Goblint detects it; I extended the test case accordingly.

**Review by jerhard — APPROVED** [review]

**Pull request merged**